File: suites/osprey-compute-kernels/reference/cases/mutual/mutual.ml
let rec is_even n = if n = 0 then true else is_odd (n - 1)
and is_odd n = if n = 0 then false else is_even (n - 1)

let () =
  let acc = ref 0 in
  for i = 1 to 129999 do
    if is_even (i mod 1000) then acc := !acc + 1
  done;
  Printf.printf "%d\n" !acc

File: suites/osprey-compute-kernels/reference/cases/mutual/mutual.rs
fn is_even(n: i64) -> bool {
    if n == 0 {
        true
    } else {
        is_odd(n - 1)
    }
}

fn is_odd(n: i64) -> bool {
    if n == 0 {
        false
    } else {
        is_even(n - 1)
    }
}

fn main() {
    let mut acc: i64 = 0;
    let mut i: i64 = 1;
    while i < 130000 {
        if is_even(i % 1000) {
            acc += 1;
        }
        i += 1;
    }
    println!("{}", acc);
}
